### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by Victoria-Moiseienko**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 .idea/*
 *.iml
-target/*
\ No newline at end of file
+target/*
+logs/*
```
Comment (AnaLanda) [inline .gitignore]: You should add an empty line here)

**Review by AnaLanda — COMMENTED** [review]
Good job! 😃

File: src/main/java/vic/cinema/Main.java [hunk 2]
```diff
@@ -18,9 +19,10 @@
 
 public class Main {
     private static Injector injector = Injector.getInstance("vic.cinema");
+    private static final Logger logger = Logger.getLogger(Main.class.getName());
```
Comment (AnaLanda) [inline src/main/java/vic/cinema/Main.java]: Try
```suggestion
    private static final Logger logger = Logger.getLogger(Main.class);
```
Should work like this as well)

File: .gitignore (hunk 1 above)
Comment (KseniiaMakarova) [reply .gitignore]: I agree

File: src/main/resources/log4j.properties [hunk 3]
```diff
@@ -0,0 +1,16 @@
+log4j.rootLogger=INFO, stdout, file
+log4j.logger.org.hibernate=INFO
+log4j.logger.org.hibernate.SQL=DEBUG
+log4j.logger.org.hibernate.type.descriptor.sql=INFO
```
Comment (KseniiaMakarova) [inline src/main/resources/log4j.properties]: do you want to save hibernate logs intentionally?

File: src/main/resources/log4j.properties [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+log4j.appender.stdout=org.apache.log4j.ConsoleAppender
+log4j.appender.stdout.Target=System.out
+log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
+log4j.appender.stdout.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n
+
+log4j.appender.file=org.apache.log4j.RollingFileAppender
+log4j.appender.file.File=C:\\logs\\cinema.log
```
Comment (KseniiaMakarova) [inline src/main/resources/log4j.properties]: let's have this file in `logs` folder that you created in your project

File: src/main/java/vic/cinema/Main.java (hunk 2 above)
Comment (KseniiaMakarova) [reply src/main/java/vic/cinema/Main.java]: I agree :)

File: src/main/java/vic/cinema/Main.java [hunk 5]
```diff
@@ -18,9 +19,10 @@
...
 
     public static void main(String[] args) throws AuthenticationException {
-        System.out.println("--- Movies ---");
+        logger.debug("--- Movies ---");
```
Comment (KseniiaMakarova) [inline src/main/java/vic/cinema/Main.java]: let's better use `info` level everywhere here

File: src/main/java/vic/cinema/Main.java [hunk 6]
```diff
@@ -68,21 +70,21 @@ public static void main(String[] args) throws AuthenticationException {
 
         authenticationService.register(user.getEmail(), user.getPassword());
         User user2 = authenticationService.login(user.getEmail(), user.getPassword());
```
Comment (KseniiaMakarova) [inline src/main/java/vic/cinema/Main.java]: let's wrap call to `login()` method in try-catch and add log.warn with the informative message in `catch` block

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
**it would be nice to add logging to DaoImpl classes** (after insert/update operations were successful)

File: src/main/java/vic/cinema/Main.java (hunk 5 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/Main.java]: Done

File: .gitignore (hunk 1 above)
Comment (Victoria-Moiseienko) [reply .gitignore]: Done

File: src/main/java/vic/cinema/Main.java (hunk 2 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/Main.java]: Done

File: src/main/resources/log4j.properties (hunk 3 above)
Comment (Victoria-Moiseienko) [reply src/main/resources/log4j.properties]: Just in case :) Removed

File: src/main/resources/log4j.properties (hunk 4 above)
Comment (Victoria-Moiseienko) [reply src/main/resources/log4j.properties]: Done

File: src/main/java/vic/cinema/Main.java (hunk 6 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/Main.java]: Done

File: src/main/java/vic/cinema/Main.java [hunk 7]
```diff
@@ -59,32 +59,37 @@ public static void main(String[] args) throws AuthenticationException {
...
                 (AuthenticationService) injector.getInstance(AuthenticationService.class);
 
         authenticationService.register(user.getEmail(), user.getPassword());
-        User user2 = authenticationService.login(user.getEmail(), user.getPassword());
-        logger.debug(user2);
+        User user2 = null;
+        try {
+            user2 = authenticationService.login(user.getEmail(), user.getPassword());
+        } catch (Exception e) {
+            logger.warn("Authentication Service: " + e);
```
Comment (KseniiaMakarova) [inline src/main/java/vic/cinema/Main.java]:
```suggestion
            logger.warn("Login failed: " + e);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
add logs to DaoImpl classes - see my comment above

File: src/main/java/vic/cinema/Main.java (hunk 7 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/Main.java]: Done

File: src/main/java/vic/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 8]
```diff
@@ -21,6 +24,7 @@ public ShoppingCart add(ShoppingCart shoppingCart) {
             transaction = session.beginTransaction();
             session.save(shoppingCart);
             transaction.commit();
+            logger.info("ShoppingCart has been added:\n" + shoppingCart);
```
Comment (KseniiaMakarova) [inline src/main/java/vic/cinema/dao/impl/ShoppingCartDaoImpl.java:27]: add this to the update() method as well

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**